### PR TITLE
Added nxos_command

### PR DESCRIPTION
**Pull request by GGabriele**

### CODE REVIEW

File: library/nxos_command.py [hunk 1]
```diff
@@ -0,0 +1,305 @@
...
+
+    device = Device(ip=host, username=username, password=password,
+                    protocol=protocol)
+
+    if isinstance(command, list):
+        module.fail_json(msg='Only single command are supported using "command". \
+            If you want to use a List, go for "command_list" instead.')
+
+    if cmd_type == 'show':
+        if command_list:
```
Comment (jedelman8) [inline library/nxos_command.py]: Sorry if I was unclear, but we can now use `command_list` even when type=show.

File: library/nxos_command.py [hunk 2]
```diff
@@ -0,0 +1,305 @@
...
+            module.fail_json(msg='Only single show commands are supported. \
+                Lists are only supported for config changes \
+                at this time.')
+
+    if cmd_type == 'config':
+        if isinstance(command_list, str):
+            module.fail_json(msg='Only Lists are supported when "command_list" \
+                is used.')
+        if text:
```
Comment (jedelman8) [inline library/nxos_command.py]: I don't think we need the `if text:` -  if users supply it, it won't be used for anything anyway.

File: library/nxos_command.py [hunk 3]
```diff
@@ -0,0 +1,305 @@
...
+    args = dict(commands=commands, text=text, cmd_type=cmd_type)
+    proposed = {}
+    changed = False
+    for param, value in args.iteritems():
+        if value is not None:
+            proposed[param] = value
+
+    if cmd_type == 'config':
+        if command:
+            if isinstance(command, str):
```
Comment (jedelman8) [inline library/nxos_command.py]: Shouldn't need this `if isinstance` here and the other one above as we can define types here:

```
command=dict(required=False, type='str'),
command_list=dict(required=False, type='list')
```

Feel free to test this too.

File: library/nxos_command.py [hunk 4]
```diff
@@ -0,0 +1,305 @@
...
+    elif cmd_type == 'config':
+        if cmds:
+            changed = True
+            result = send_command(device, cmds)
+
+    results = {}
+    results['changed'] = changed
+    results['proposed'] = proposed
+    results['commands'] = [commands]
+    results['result'] = result
```
Comment (jedelman8) [inline library/nxos_command.py]: change this key to `response`

File: library/nxos_command.py [hunk 5]
```diff
@@ -0,0 +1,305 @@
...
+
+    elif cmd_type == 'config':
+        if cmds:
+            changed = True
+            result = send_command(device, cmds)
+
+    results = {}
+    results['changed'] = changed
+    results['proposed'] = proposed
+    results['commands'] = [commands]
```
Comment (jedelman8) [inline library/nxos_command.py]: isn't this creating a list of lists for `commands`? We don't need that...too messy.

Instead, just `cmds` and return the command string from `command_list_to_string` - you'll still need to convert `command` to a list before calling that function though :)

File: library/nxos_command.py (hunk 3 above)
Comment (GGabriele) [inline library/nxos_command.py]: At first I did just like you said but I had some issue with this. If I do so, and use a pb like the following, I should receive an error:

```
- name: "CONFIGURE LOOPBACK INTERFACE"
    nxos_command: 
      host: "{{ inventory_hostname }}"
      type: config
      command: ['interface lo11','ip add 11.11.11.11 255.255.255.0']
    register: data

  - debug: var=data
```

..what happens, instead, is that my `command`  (which should be a string and here is a list instead) is casted as a string as well:

```
"command": "['interface lo11', 'ip add 11.11.11.11 255.255.255.0']" 
```

Similar thing would happen with `command_list: 'interface lo11, ip add 11.11.11.11 255.255.255.0'` which will be automatically converted to list:

```
"command_list": [
                "interface lo11", 
                " ip add 11.11.11.11 255.255.255.0"
            ]
```

Anyway, this last case is not a big problem since we can still use this

**Comment by jedelman8** [comment]
Can you make these changes before making new PRs?

**Comment by GGabriele** [comment]
Sure. chages are ready...but please take a look at my reply on the if `isinstance` section above.

File: library/nxos_command.py (hunk 3 above)
Comment (jedelman8) [inline library/nxos_command.py]: Gotcha...okay, keep the isinstance/module fail checks, but they should be generic...shouldn't matter if it's for show/config commands.

File: library/nxos_command.py [hunk 6]
```diff
@@ -0,0 +1,301 @@
...
+        if isinstance(command_list, list):
+            cmds = command_list_to_string(commands)
+        else:
+            module.fail_json(msg='Only Lists are supported using "command_list". \
+                If you want to use a String, go for "command" instead.')
+
+    response = []
+    if cmd_type == 'show':
+        if command:
+            splitted_command = command.split(',')
```
Comment (jedelman8) [inline library/nxos_command.py]: Why is this here?  Didn't we say weren't supporting comma delimited commands?  We shouldn't need this as the user can still do `interface lo13 ; ip add 13.13.13.13 255.255.255.0` in a playbook.  Ensure that works :)
